ts/src/pro/binance.ts

```diff
@@ -283,6 +294,81 @@ export default class binance extends binanceRest {
         return orderbook.limit ();
     }
 
+    async fetchOrderBookWs (symbol: string, limit: Int = undefined, params = {}): Promise<OrderBook> {
+        /**
+         * @method
+         * @name binance#fetchOrderBookWs
+         * @description fetches information on open orders with bid (buy) and ask (sell) prices, volumes and other data
+         * @see https://binance-docs.github.io/apidocs/futures/en/#order-book-2
+         * @param {string} symbol unified symbol of the market to fetch the order book for
+         * @param {int} [limit] the maximum amount of order book entries to return
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @returns {object} A dictionary of [order book structures]{@link https://docs.ccxt.com/#/?id=order-book-structure} indexed by market symbols
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const payload = {
+            'symbol': market['id'],
+        };
+        if (limit !== undefined) {
+            payload['limit'] = limit;
+        }
+        const marketType = this.getMarketType ('fetchOrderBookWs', market, params);
+        if (marketType !== 'future') {
```

Comment:
according to our terminology, perps are called swaps, so using `future` in the exception message might be a bit ambiguous